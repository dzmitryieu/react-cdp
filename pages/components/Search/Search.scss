@import "../../utils/constants.scss";

.searchWrapper {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
}

.search {
  color: white;
  border: none;
  border-bottom: 2px solid $light-red;
  font-size: 14px;
  font-family: $helvetica;
  height: 33px;
  width: 100%;
  background-color: black;
  background-image: url(../../assets/images/search-icon.png);
  background-repeat: no-repeat;
  background-position: right 10px center;
  background-size: 20px;
  outline: none;
}

.search::placeholder {
  color: white;
  font-size: 14px;
  opacity: 1;
}

.searchText {
  font-size: 20px;
  color: white;
  text-transform: uppercase;
  font-family: $helvetica;
}

.searchTitle {
  color: white;
  font-family: $helvetica;
}

.buttonsWrapper {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.btnWrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}
