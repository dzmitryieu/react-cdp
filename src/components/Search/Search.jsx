import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import styles from './Search.scss';
import { ENTER_KEY } from '../../utils/consts';

const searchItems = [
  {
    text: 'title',
    field: 'title',
  },
  {
    text: 'genre',
    field: 'genres',
  },
];

class Search extends React.Component {
  state = {
    word: '',
    searchBy: 'title',
    navigationTitle: 'title',
  }

  componentDidMount() {
    document.addEventListener('keypress', this.onSubmitHandler);
  }

  componentWillUnmount() {
    document.removeEventListener('keypress', this.onSubmitHandler);
  }

  onSubmitHandler = (e) => {
    if (e.keyCode === ENTER_KEY) {
      this.onClickHandler();
    }
  }

  onChangeHandler = (e) => {
    this.setState({
      word: e.target.value,
    });
  }

  onClickHandler = () => {
    const { word, searchBy, navigationTitle } = this.state;
    const { searchFilm } = this.props;
    const searchWord = word.toLowerCase();
    searchFilm({ word: searchWord, searchBy, navigationTitle });
    this.setState({
      word: '',
    });
  };

  onSearchByHandler = (e) => {
    this.setState({
      searchBy: e.target.dataset.attr,
      navigationTitle: searchItems.find(el => el.field === e.target.dataset.attr).text,
    });
  }

  render() {
    const { word, searchBy } = this.state;
    return (
      <div className={styles.searchWrapper}>
        <p className={styles.searchText}>find your movie</p>
        <input
          type="text"
          className={styles.search}
          placeholder="search film"
          value={word}
          onChange={this.onChangeHandler}
        />
        <div className={styles.buttonsWrapper}>
          <div className={styles.btnWrapper}>
            <h3 className={styles.searchTitle}>SEARCH BY</h3>
            {
              searchItems.map(item => (
                <Button
                  key={item.text}
                  type={searchBy === item.field ? 'red' : 'black'}
                  text={item.text}
                  dataAttr={item.field}
                  onClick={this.onSearchByHandler}
                />
              ))
            }
          </div>
          <Button
            type="red"
            text="SEARCH"
            isMain
            onClick={this.onClickHandler}
          />
        </div>
      </div>
    );
  }
}

Search.propTypes = {
  searchFilm: PropTypes.func.isRequired,
};

export default Search;
