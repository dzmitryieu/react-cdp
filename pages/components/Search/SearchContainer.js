import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Search from './Search';
import { fetchFilmByRoute } from '../../../src/modules/film/filmActions';
import { fetchAllFilms } from '../../../src/modules/films/filmsActions';
import {
  searchFilm,
  cleanSearchField,
} from '../../../src/modules/searchFilms/searchFilmsActions';
import { getSearchingWord } from '../../../src/modules/searchFilms/searchFilmsSelectors';


const mapStateToProps = state => ({
  searchedWord: getSearchingWord(state),
});

const mapDispatchToProps = {
  searchFilm,
  cleanSearchField,
  fetchAllFilms,
  fetchFilmByRoute,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Search));
