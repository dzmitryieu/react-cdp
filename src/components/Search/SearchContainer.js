import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Search from './Search';
import { fetchFilmByRoute } from '../../modules/film/filmActions';
import { fetchAllFilms } from '../../modules/films/filmsActions';
import {
  searchFilm,
  cleanSearchField,
} from '../../modules/searchFilms/searchFilmsActions';
import { getSearchingWord } from '../../modules/searchFilms/searchFilmsSelectors';


const mapStateToProps = state => ({
  searchedWord: getSearchingWord(state),
});

const mapDispatchToProps = {
  searchFilm,
  cleanSearchField,
  fetchAllFilms,
  fetchFilmByRoute,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Search));
