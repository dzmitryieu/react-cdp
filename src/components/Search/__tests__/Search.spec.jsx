import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Search from '../index';

const shallow = new ShallowRenderer();

describe('Search Snapshot', () => {
  test('render', () => {
    const component = shallow.render(
      <Search />,
    );
    expect(component).toMatchSnapshot();
  });
});
