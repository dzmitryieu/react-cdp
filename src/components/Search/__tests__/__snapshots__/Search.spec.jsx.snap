// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot render 1`] = `
<div
  className="searchWrapper"
>
  <p
    className="searchText"
  >
    find your movie
  </p>
  <input
    className="search"
    onChange={[Function]}
    placeholder="search film"
    type="text"
    value=""
  />
  <div
    className="buttonsWrapper"
  >
    <div
      className="btnWrapper"
    >
      <h3
        className="searchTitle"
      >
        SEARCH BY
      </h3>
      <Button
        className={null}
        dataAttr="title"
        id={null}
        isMain={null}
        onClick={[Function]}
        text="title"
        type="red"
      />
      <Button
        className={null}
        dataAttr="genres"
        id={null}
        isMain={null}
        onClick={[Function]}
        text="genre"
        type="black"
      />
    </div>
    <Button
      className={null}
      dataAttr={null}
      id={null}
      isMain={true}
      onClick={[Function]}
      text="SEARCH"
      type="red"
    />
  </div>
</div>
`;

exports[`Search Snapshot with store render 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "notPersistedStore": Object {
          "searchBy": "title",
          "searchedWord": "word",
        },
      },
    }
  }
>
  <withRouter(Connect(Search)) />
</Context.Provider>
`;
